src/imperative/cached_op.cc

```diff
@@ -674,7 +674,7 @@ OpStatePtr CachedOp::StaticForward(
   std::lock_guard<std::mutex> lock(state.mutex);
 
   bool match = SetForwardGraph(&state.info, recording, inputs);
-  match = match && state.recording != recording;
+  match = match && state.recording == recording;
```

Comment:
Test?

Reply:
It only affects performance. hard to test

Reply:
Well this is a logical expression and it seems like it did something that it was not supposed it was to do due to a wrong boolean expression.

Reply:
cache was invalidated unnecessarily